packages/graphql-language-service-interface/src/fileUtils.ts

```diff
@@ -0,0 +1,57 @@
+function getExtension (pathParts: string[]): string | null {
+  return pathParts.pop() || null
+}
+
+function removeExtension  (filePath: string, pathParts: string[]): string {
+  // if there's a file extension
+  if (pathParts.length > 1) {
+    // pop the extension from the array
+    pathParts.pop() as string
+    return pathParts.join('.')
+  }
+  // if there is no file extension
+  else {
+    return filePath
+  }
+}
+
+// these make webpack happy
+
+export function resolveFile (filePath: string) {
+  const pathParts = filePath.split('.')
+  const extension = getExtension(pathParts)
```

Comment:
This does not seem a good way of getting the extension. For example it would mess up on the path `/home/me/my.work/foo` or `/path/to/something/../otherthing/foo` and it would be expensive on a path with lots of dots in.

Reply:
yeah good point. i was hoping just popping the last item would work, but that won't work with extensionless paths. ill write some unit tests that take this into account and rewrite accordingly. I want to use the `path` module because it has great utilities for this, but we are trying to steer GLS-interface away from being node dependent, since we need it to be able to be mounted in the browser, thus fully isomorphic in the coming months.

Reply:
I think webpack at least has polyfills for the `path` module. Likely other bundlers do too?